addon-test-support/@ember/test-helpers/dom/type-in.ts

```diff
@@ -94,7 +95,20 @@ function keyEntry(element: FormControl, character: string): () => void {
       .then(() => __triggerKeyEvent__(element, 'keydown', characterKey, options))
       .then(() => __triggerKeyEvent__(element, 'keypress', characterKey, options))
       .then(() => {
-        element.value = element.value + character;
+        const newValue = element.value + character;
+        const maxlength = element.getAttribute('maxlength');
+        const shouldTruncate =
+          isMaxLengthConstrained(element) &&
+          maxlength &&
+          newValue &&
+          newValue.length > Number(maxlength);
+        if (shouldTruncate) {
```

Comment:
The variable name looks obsolete.

When looking at it and at the condition above, I'm wondering if we can hide this logic into a standalone util, like:

```ts
/**
 * @throws if `value` exceeds the `maxlength`(if specified)
 */
function guardForMaxlength(element: FormControl, value: string): void;
```

I believe it could reduce duplication a bit.

Reply:
yeah that made sense. `isMaxLengthConstrained` then became just an implementation detail of `guardForMaxlength`.